tests/phpunit/tests/html-api/wpHtmlSupportRequiredOpenElements.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Unit tests for the HTML API indicating that changes are needed to the
+ * WP_HTML_Open_Elements class before specific features are added to the API.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ */
+
+/**
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_wpHtmlSupportRequiredOpenElements extends WP_UnitTestCase {
+	private function ensure_support_is_added_everywhere( $tag_name ) {
+		$p = WP_HTML_Processor::createFragment( "<$tag_name>" );
+
+		$this->assertFalse( $p->step(), "Must support terminating elements in specific scope check before adding support for the {$tag_name} element." );
```

Comment:
@hellofromtonya Correct me if wrong, but should assertions only be performed inside the test methods / closures within test methods?

Reply:
so this would be a good place to examine, as this is a fairly novel kind of test file. I have explored a few situations. the point here is to actively warn someone who wants to add additional support to the HTML parser class.

given that we have hundreds of tests that all do the same thing, I figured a minor abstraction here would be valuable to have, despite generally agreeing that test abstractions are things to be wary of.

if there's a way to ensure this won't be called outside of a test method that works; this was my intent on not naming it with the `test_` prefix.

Reply:
the more I look into this the more I'd rather special-case this from the linting rules than create a new test assertion. creating a new assertion is going to pull in more complexity and make things more convoluted than they already are to debug and understand. having this single function, while different than in other cases, at least is self-contained and straightforward.

so I'd rather we do whatever we need to do to make sure this doesn't get abstracted into something much greater in scope than is required and allow this unique class to be unique.

more important than enforcing this styling rule, I'd love some feedback on automating the support rules, to ensure that nobody unintentionally _partially_ adds support for HTML tags that require additional support. for example, we don't want someone to simply add a new `BUTTON` clause to the big switch statement in the html processor. if they do that, they will subvert other rules that have to be added to the adoption agency algorithm.